patroni/dcs/etcd.py

```diff
@@ -480,7 +502,7 @@ def _handle_exception(self, e, name='', do_sleep=False, raise_ex=None):
         if isinstance(raise_ex, Exception):
             raise raise_ex
 
-    def _run_and_handle_exceptions(self, method, *args, **kwargs):
+    def _run_and_handle_exceptions(self, method: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
         retry = kwargs.pop('retry', self.retry)
         try:
             return retry(method, *args, **kwargs) if retry else method(*args, **kwargs)
```

Comment:
The signature of the retry method doesn't match self.retry, which doesn't match the `__call__` of Retry which as `func` as the first argument.